### ISSUE TITLE
Reason for AMS2 not working

### ISSUE THREAD

**Issue by Jabbah** [fholger/openvr_fsr#15]
@fholger AMS2 (and PC2) use an odd very old version of OpenVR - 0.9.12 (Not even RaceRoom is that old!) It's very picky about the version, even openvr 0.9.13 won't work with it (I think because the IVR version strings are hard coded in the DLL so you can't request a different version). Unfortunately there were some pretty major changes to the interface after that so it might be pretty difficult to get this working properly with things as they are.

I got it working by replacing the openvr.h, libs, and dlls with version 0.9.12 and copying the COpenVRContext and supporting functions from the new openvr.h to openvr_api_public.cpp. Not sure if that's all necessary, but if it is there doesn't look to be a way to have one version of this that works for all. Any ideas?

**Comment by fholger** [comment]
Yeah. Use injection through a different DLL (e.g. d3d11.dll or dxgi.dll). But that requires a bit of refactoring to get working, so no short-term fix, unfortunately.

**Comment by FDanner** [comment]
Someone posted a modification here that works for AMS2.
https://forum.reizastudios.com/threads/fsr-for-openvr.19036/#post-143792
But I tested it and the results are about the same as lowering the resolution by other means.

**Comment by wotever** [comment]
> Someone posted a modification here that works for AMS2.
> https://forum.reizastudios.com/threads/fsr-for-openvr.19036/#post-143792
> But I tested it and the results are about the same as lowering the resolution by other means.

does it not see a benefit from the sharpening part at least? like fake supersampling the image and increasing sharpness etc.
i feel like dirt rally 2.0 looks a bit better and sharper using this mod, but eh, i have not done enough testing. anyway, are you saying that there was no performance or visual benefit benefit?

**Comment by Jabbah** [comment]
> does it not see a benefit from the sharpening part at least? like fake supersampling the image and increasing sharpness etc.
> i feel like dirt rally 2.0 looks a bit better and sharper using this mod, but eh, i have not done enough testing. anyway, are you saying that there was no performance or visual benefit benefit?

I did the AMS2 version, but after lots of testing and feedback it just didn't seem worth the effort to get to maintain it. The problem is that SteamVR already allows upsampling by turning down the resolution percentage in SteamVR. AMS2 has config to provide sharpening and the results from that are pretty close to what you get from FSR but perform slightly better. IMHO it just doesn't seem like AMS2 (or ACC for that matter) benefit much from FSR compared to tuning StemVR res and in game sharpening.

**Comment by reptilexcq** [comment]
It crashes for me every time I try to load up the game in VR.  I have Pimax 8KX.  I would love to test it out but I guess it's not working for the Slightly Mad Studio engine.

**Comment by Mavy1704** [comment]
Unfortunately for me it always crashes :-(
